virtcontainers/qemu_amd64.go

```diff
@@ -83,12 +86,12 @@ func maxQemuVCPUs() uint32 {
 	return uint32(240)
 }
 
-func newQemuArch(machineType string) qemuArch {
+func newQemuArch(machineType string, rootImage bool) qemuArch {
```

Comment:
Could you modify the function so that we directly pass the hypervisor configuration instead. This will prevent from adding some specific parameters like `rootImage bool` and the check will be performed from each implementation if needed:
```go
func newQemuArch(config HypervisorConfig) qemuArch {
        ...
        if config.ImagePath != "" {
        ...
}
```